src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -31,57 +32,131 @@ public partial class TestAgency : ITestAgency, IService
+        }
+
+        #endregion
+
+        #region ITestAgentProvider Implementation
+
+        /// <summary>
+        /// Returns true if an agent can be found, which is suitable
+        /// for running the provided test package.
+        /// </summary>
+        /// <param name="package">A TestPackage</param>
+        public bool IsAgentAvailable(TestPackage package)
+        {
+            foreach (var launcher in _launchers)
+                if (launcher.CanCreateAgent(package))
+                    return true;
+
+            return false;
+        }
+
+        /// <summary>
+        /// Return an agent, which best matches the criteria defined
+        /// in a TestPackage.
+        /// </summary>
+        /// <param name="package">The test package to be run</param>
+        /// <returns>An ITestAgent</returns>
+        /// <exception cref="ArgumentException">If no agent is available.</exception>
+        public ITestAgent GetAgent(TestPackage package)
+        {
             // Target Runtime must be specified by this point
             string runtimeSetting = package.GetSetting(EnginePackageSettings.TargetRuntimeFramework, string.Empty);
             Guard.OperationValid(runtimeSetting.Length > 0, "LaunchAgentProcess called with no runtime specified");
             bool runAsX86 = package.GetSetting(EnginePackageSettings.RunAsX86, false);
 
-            // If target runtime is not available, something went wrong earlier.
-            // We list all available frameworks to use in debugging.
             var targetRuntime = RuntimeFramework.Parse(runtimeSetting);
-            if (!_runtimeService.IsAvailable(targetRuntime.Id, runAsX86))
-            {
-                var runtimes = runAsX86 ? _availableRuntimeService.AvailableX86Runtimes : _availableRuntimeService.AvailableRuntimes;
-                string msg = $"The {targetRuntime} framework is not available.\r\nAvailable frameworks:";
-                foreach (var runtime in runtimes)
-                    msg += $" {runtime}";
-                throw new ArgumentException(msg);
-            }
-
             var agentId = Guid.NewGuid();
-            var agentProcess = new AgentProcess(this, package, agentId);
-            var agentName = targetRuntime.Id + (runAsX86 ? "-x86" : string.Empty) + "-agent";
-            package.AddSetting("SelectedAgentName", agentName);
+            string agencyUrl = targetRuntime.FrameworkName.Identifier == ".NETFramework" ? RemotingUrl : TcpEndPoint;
+            var agentProcess = CreateAgentProcess(agentId, agencyUrl, package);
 
-            agentProcess.Exited += (sender, e) => OnAgentExit((Process)sender!, agentId);
+            agentProcess.Exited += (sender, e) => OnAgentExit(agentProcess);
 
             agentProcess.Start();
-            log.Debug("Launched Agent process {0} - see nunit-agent_{0}.log", agentProcess.Id);
+            log.Debug("Launched Agent process {0} - see testcentric-agent_{0}.log", agentProcess.Id);
```

Comment:
You did mention you got some of this from _testcentric_

Reply:
Are you seeing logs created with that name? If so, that's what I should fix.

Reply:
Even if not, you point users to look at a non-existing one.

Reply:
Oh, sorry. I misread that was a change suggestion nunit => testcentric. I'll fix.